src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -98,29 +106,106 @@ public async Task<AuthenticationResult> RunAsync(CancellationToken cancellationT
                     LogMetricsFromAuthResult(authenticationResult, AuthenticationRequestParameters.RequestContext.Logger);
                     LogSuccessfulTelemetryToClient(authenticationResult, telemetryEventDetails, telemetryClients);
 
+                    OpenTelemetryClient.LogSuccessMetrics(
+                        ServiceBundle.PlatformProxy.GetProductName(), 
+                        AuthenticationRequestParameters.AppConfig.ClientId, 
+                        authenticationResult.AuthenticationResultMetadata, 
+                        GetCacheLevel(authenticationResult).ToString());
+
+                    LogMsalSuccessTelemetryToOTel(authenticationResult);
+
                     return authenticationResult;
                 }
                 catch (MsalException ex)
                 {
                     apiEvent.ApiErrorCode = ex.ErrorCode;
                     AuthenticationRequestParameters.RequestContext.Logger.ErrorPii(ex);
                     LogMsalErrorTelemetryToClient(ex, telemetryEventDetails, telemetryClients);
+
+                    OpenTelemetryClient.LogToFailureCounter(
```

Comment:
Don't you want to use the same name (Metric instead of Counter)?

Reply:
The counters are built using the meter. So all the counters and histograms that we have currently will be registered to the meter `ID4S_Msal.Net.Meter`. So the meter will have a success counter, failure counter and histograms based on total duration, duration in cache and duration in http.